src/galaxy/StarSystemGenerator.cpp

```diff
@@ -1290,13 +1290,9 @@ SystemBody* StarSystemRandomGenerator::MakeGravPointForBodies(RefCountedPtr<Star
 	return grav;
 }
 
-std::string StarSystemRandomGenerator::GetStarNameFromId(const int id)
+char StarSystemRandomGenerator::GetStarSuffixFromOffset(const int offsetId) const
 {
-	// UGLY HACK
-	assert((id >= 0) && (id < 26));
```

Comment:
I'd suggest not removing this assert.

Silently ignoring wrong values would break unique star names.
offsetId < 0 -> "SystemName A" for any negative value
offsetId >= 26 -> "SystemName Z" for any value

Reply:
restored it